### ISSUE TITLE
CDC panic: The CRTs must be greater than the resolvedTs

### ISSUE THREAD

**Issue by Tammyxia** [pingcap/ticdc#1576]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Create tidb cluster + 3x cdc server binary version + 1x changfeed, sysben insert data to upstream
- Restart all pd: tiup cluster restart tidb-test -R pd
- Aftert 5 minutes Find cdc server can recovery and changfeed can continue to sync.
- Repeat step 2,3 per 10 minutes, find cdc oom killed as a known issue.
- Start CDC server aggain:  $ cdc server --pd=http://34.217.25.239:2379 --advertise-addr=52.24.165.124:8300 --addr=0.0.0.0:8300 --log-file=cdc.log --sort-dir=/tmp/cdc-lx

2. What did you expect to see?

3. What did you see instead?
cdc server exit with error message:
- {"level":"warn","ts":"2021-03-30T03:30:33.586Z","caller":"clientv3/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"endpoint://client-cf42803b-6595-4cba-aa32-069309f0e633/34.217.25.239:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
panic: The CRTs must be greater than the resolvedTs

goroutine 48868 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xd1f5ae1c80, 0xcf90f21d00, 0x4, 0x4)
        go.uber.org/zap@v1.16.0/zapcore/entry.go:234 +0x567
go.uber.org/zap.(*Logger).Panic(0xc3fc2da8a0, 0x2c78d20, 0x2c, 0xcf90f21d00, 0x4, 0x4)
        go.uber.org/zap@v1.16.0/logger.go:226 +0x7f
github.com/pingcap/log.Panic(0x2c78d20, 0x2c, 0xcf90f21d00, 0x4, 0x4)
        github.com/pingcap/log@v0.0.0-20201112100606-8f1e84a3abc8/global.go:50 +0xff
github.com/pingcap/ticdc/cdc/puller.(*pullerImpl).Run.func5.1(0xd23faa1180, 0xd23faa1180, 0x1b)
        github.com/pingcap/ticdc@/cdc/puller/puller.go:194 +0x39a
github.com/pingcap/ticdc/cdc/puller.(*pullerImpl).Run.func5(0xc3d6ad2768, 0x0)
        github.com/pingcap/ticdc@/cdc/puller/puller.go:253 +0x33f
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc3d54a04e0, 0xc3d6aa40c0)
        golang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync@v0.0.0-20201020160332-67f06af15bc9/errgroup/errgroup.go:54 +0x66

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
        Release Version: v5.0.0
        Edition: Community
        Git Commit Hash: b18eae98087148ad967d7fcbf6586ed7b9607c4b
        Git Branch: heads/refs/tags/v5.0.0
        UTC Build Time: 2021-03-24 15:31:59
        GoVersion: go1.13
        Race Enabled: false
        TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
        Check Table Before Drop: false |

        ```

    - TiCDC version (execute `cdc version`):

        ```
        Release Version: v5.0.0-nightly-15-g8968aae
        Git Commit Hash: 8968aae98bc9832afb0c484a9e01bb8e0c665ba7
        Git Branch: HEAD
        UTC Build Time: 2021-03-29 08:31:59
        ```

**Comment by amyangfei** [comment]
```
cdc-2021-03-30T03-31-18.852.log:[2021/03/30 03:28:35.151 +00:00] [INFO] [region_range_lock.go:218] ["range locked"] [lockID=387] [regionID=5825941] [startKey=748000000000000fff245f728000000000ff019fae0000000000fa] [endKey=748000000000000fff245f728000000000ff01e1390000000000fa] [checkpointTs=423906485846933529]
cdc-2021-03-30T03-31-18.852.log:[2021/03/30 03:28:35.151 +00:00] [INFO] [client.go:817] ["start new request"] [request="{\"header\":{\"cluster_id\":6942784005347452364,\"ticdc_version\":\"5.0.0-nightly\"},\"region_id\":5825941,\"region_epoch\":{\"conf_ver\":5,\"version\":1008},\"checkpoint_ts\":423906485846933529,\"start_key\":\"dIAAAAAAAA//JF9ygAAAAAD/AZ+uAAAAAAD6\",\"end_key\":\"dIAAAAAAAA//JF9ygAAAAAD/AeE5AAAAAAD6\",\"request_id\":117402,\"extra_op\":1,\"Request\":null}"] [addr=44.234.9.155:20160]
```

**Comment by amyangfei** [comment]
Still under investigation, haven't reproduce

**Comment by lonng** [comment]
I change the severity from critical to major because this bug can't be reproduced.

**Comment by amyangfei** [comment]
another case [cdc-2021-06-10T03-33-13.437.log.tar.gz](https://github.com/pingcap/ticdc/files/6628261/cdc-2021-06-10T03-33-13.437.log.tar.gz)

```
[2021/06/10 01:47:48.893 +08:00] [INFO] [version.go:47] ["Welcome to Change Data Capture (CDC)"] [release-version=v5.0.0-master-dirty] [git-hash=8bcab34127d7268e83d39cc677fcee9dcab44be3] [git-branch=fix_etcd_txn_too_big] [utc-build-time="2021-06-09 10:13:24"] [go-version="go version go1.16.4 linux/amd64"] [failpoint-build=false]
```

**Comment by amyangfei** [comment]
> another case [cdc-2021-06-10T03-33-13.437.log.tar.gz](https://github.com/pingcap/ticdc/files/6628261/cdc-2021-06-10T03-33-13.437.log.tar.gz)
> 
> ```
> [2021/06/10 01:47:48.893 +08:00] [INFO] [version.go:47] ["Welcome to Change Data Capture (CDC)"] [release-version=v5.0.0-master-dirty] [git-hash=8bcab34127d7268e83d39cc677fcee9dcab44be3] [git-branch=fix_etcd_txn_too_big] [utc-build-time="2021-06-09 10:13:24"] [go-version="go version go1.16.4 linux/amd64"] [failpoint-build=false]
> ```

After investigation, this case tends to be the same cause with bug found in the original issue.

A potential cause is a table merges more region beyond its original region range, which causes resolved fallen back

**Comment by amyangfei** [comment]
This bug has been fixed by #2038 and #2077
